File: OLD_SRC_2/util/TextUtil.java
package com.sleepwalker.sleeplib.util;

import net.minecraft.util.text.IFormattableTextComponent;
import net.minecraft.util.text.StringTextComponent;

import javax.annotation.Nonnull;

public final class TextUtil {

    @Nonnull
    public static IFormattableTextComponent str(@Nonnull String text){
        return new StringTextComponent(text.replace("&", "§"));
    }

    @Nonnull
    public static IFormattableTextComponent str(@Nonnull String text, Object... args){
        return new StringTextComponent(String.format(text, args).replace("&", "§"));
    }
}

File: src/main/kotlin/com/sleepwalker/sleeplib/util/TextUtil.kt
package com.sleepwalker.sleeplib.util

import net.minecraft.util.text.IFormattableTextComponent
import net.minecraft.util.text.StringTextComponent

fun String.literal(): IFormattableTextComponent = StringTextComponent(this.replace("&", "§"))
fun String.literal(vararg args: Any): IFormattableTextComponent = StringTextComponent(String.format(this, args).replace("&", "§"))